openassessment/assessment/serializers/training.py

```diff
@@ -181,9 +181,10 @@ def deserialize_training_examples(examples, rubric_dict):
                 example = TrainingExample.objects.get(content_hash=content_hash)
             except TrainingExample.DoesNotExist:
                 try:
-                    example = TrainingExample.create_example(
-                        example_dict['answer'], example_dict['options_selected'], rubric
-                    )
+                    with transaction.atomic():
+                        example = TrainingExample.create_example(
```

Comment:
It would be preferable to more closely follow the django model api and override get_or_create and create method to push some of this logic down into the model. We are doing things here that the serializer shouldn't need to worry about.